### PR TITLE
Fix #1272: Add TraceID/SpanID to Monitoring for Enhanced Observability

### LINKED ISSUE #1272: Add TraceID/SpanID to Monitoring for Enhanced Observability

### PR DESCRIPTION
**Pull request by jandusil**

### CODE REVIEW

File: powerauth-admin/src/main/resources/application.properties [hunk 1]
```diff
@@ -51,6 +51,7 @@ spring.jpa.open-in-view=false
 management.health.ldap.enabled=false
 
 # Monitoring
+management.tracing.sampling.probability=1.0
```
Comment (banterCZ) [inline powerauth-admin/src/main/resources/application.properties:54]: @zcgandcomp Should it be turned on by default?

**Review by banterCZ — APPROVED** [review]
Looks good. I am just not sure about the default configuration.

File: powerauth-admin/src/main/resources/application.properties (hunk 1 above)
Comment (zcgandcomp) [reply powerauth-admin/src/main/resources/application.properties:54]: yes we can have this turned on by default, but provide a way how to configure it. ie. 1 is default, but there has to be a variable to override it.

File: powerauth-java-server/src/main/resources/application.properties [hunk 2]
```diff
@@ -125,7 +125,4 @@ powerauth.service.correlation-header.value.validation-regexp=[a-zA-Z0-9\\-]{8,10
 powerauth.service.proximity-check.otp.length=8
 
 # Monitoring
-#management.endpoint.metrics.enabled=true
```
Comment (banterCZ) [inline powerauth-java-server/src/main/resources/application.properties:128]: Why was this removed?

File: powerauth-admin/src/main/java/io/getlime/security/app/admin/configuration/ApplicationConfiguration.java [hunk 3]
```diff
@@ -55,6 +55,12 @@ public class ApplicationConfiguration {
     @Value("${powerauth.admin.service.applicationEnvironment}")
     private String applicationEnvironment;
 
+    /**
+     * Specifies the proportion of requests that are sampled for tracing.
+     */
+    @Value("${management.tracing.sampling.probability}")
+    private double tracingSamplingProbability;
```
Comment (banterCZ) [inline powerauth-admin/src/main/java/io/getlime/security/app/admin/configuration/ApplicationConfiguration.java:62]: Since we do not use this value ourselves, declaring it in the configuration class is redundant.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthServiceConfiguration.java [hunk 4]
```diff
@@ -264,6 +264,12 @@ public class PowerAuthServiceConfiguration {
     @Value("${powerauth.service.proximity-check.otp.length:8}")
     private int proximityCheckOtpLength;
 
+    /**
+     * Specifies the proportion of requests that are sampled for tracing.
+     */
+    @Value("${management.tracing.sampling.probability}")
+    private double tracingSamplingProbability;
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/configuration/PowerAuthServiceConfiguration.java:271]: dtto

File: powerauth-admin/src/main/java/io/getlime/security/app/admin/configuration/ApplicationConfiguration.java (hunk 3 above)
Comment (jandusil) [reply powerauth-admin/src/main/java/io/getlime/security/app/admin/configuration/ApplicationConfiguration.java:62]: @banterCZ
I thought, the goal was to add variable to override default value, no? 

> yes we can have this turned on by default, but provide a way how to configure it. ie. 1 is default, but there has to be a variable to override it.

**Review by banterCZ — APPROVED** [review]

**Comment by jandusil** [comment]
In the end, we are not gonna mention the `management.tracing.enabled` property as it is intended more for testing.
[https://github.com/spring-projects/spring-boot/issues/38625](url)

File: docs/Configuration-Properties.md [hunk 5]
```diff
@@ -71,6 +71,9 @@ The PowerAuth Server uses the following public configuration properties:
 
 
 ## Monitoring and Observability
+| Property                                  | Default | Note                                                              |
+|-------------------------------------------|---------|-------------------------------------------------------------------|
+| `management.tracing.sampling.probability` | `1.0`   | Specifies the proportion of requests that are sampled for tracing |
```
Comment (zcgandcomp) [inline docs/Configuration-Properties.md:76]: Explicitly state that value 0 disables the tracing. Otherwise, fine with me.

**Review by zcgandcomp — CHANGES_REQUESTED** [review]
Update to the docs

File: docs/Configuration-Properties.md [hunk 6]
```diff
@@ -71,6 +71,9 @@ The PowerAuth Server uses the following public configuration properties:
 
 
 ## Monitoring and Observability
+| Property                                  | Default | Note                                                                                                                                                                       |
+|-------------------------------------------|---------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `management.tracing.sampling.probability` | `1.0`   | Specifies the proportion of requests that are sampled for tracing. A value of 1.0 means that 100% of requests are sampled, while a value of 0 effectively disables tracing |
```
Comment (romanstrobl) [inline docs/Configuration-Properties.md:76]: Missing period at the end of second sentence.

**Review by romanstrobl — APPROVED** [review]
Looks OK.

**Review by zcgandcomp — APPROVED** [review]
Looks nice.
